Library/Homebrew/compilers.rb

```diff
@@ -133,6 +133,19 @@ def find_compiler
     compilers.each do |compiler|
       case compiler
       when :gnu
+        unless OS.mac?
```

Comment:
I hope we can change this one day. With Ubuntu 18.04 gcc will be 7.x anyway, so everything will change again. I would be in favour to have the gcc formula and aliases like upstream. `gcc` should just be the lost recent gcc, because the current state is ... surprising for newcomers.

Reply:
In fact, I think it can be argued that Mac should use this logic as well. That is to check default gcc formula first.